src/twisted/python/reflect.py

```diff
@@ -134,7 +134,8 @@ def accumulateMethods(obj, dict, prefix='', curClass=None):
     if not curClass:
         curClass = obj.__class__
     for base in curClass.__bases__:
-        accumulateMethods(obj, dict, prefix, base)
+        if base is not object:
```

Comment:
OK, I did some tests on CPython 2.7 and CPython 3.6 and this change works.
It needs a comment though.